snapcraft/tests/test_file_utils.py

```diff
@@ -64,6 +64,25 @@ def test_replace_in_file(self):
         with open(self.file_path, 'r') as f:
             self.assertEqual(f.read(), self.expected)
 
+    def test_replace_in_file_with_permission_error(self):
+        os.makedirs('bin')
+        file_info = {
+            'path': os.path.join('bin', 'readonly'),
+            'contents': '#!/foo/bar/baz/python',
+            'expected': '#!/foo/bar/baz/python',
+        }
+        with open(file_info['path'], 'w') as f:
+            f.write(file_info['contents'])
+
+        with mock.patch('snapcraft.file_utils.open',
+                        side_effect=PermissionError('')):
```

Comment:
um, right, because they run as sudo. We should run as a normal user, but ugh docker.
A comment in the code explaining this would be appreciated.